pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -550,6 +550,62 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 	}
 }
 
+// PublishAppInstMetaDataToZedCloud is called when an appInst reports its Metadata to EVE.
+// AppInst metadata is relayed to the controller to be processed further.
+func PublishAppInstMetaDataToZedCloud(ctx *zedagentContext, appInstID string, appInstMetadata *types.AppInstMetaData, iteration int) {
+	log.Functionf("PublishAppInstMetaDataToZedCloud: appInstID: %v, metadataType: %v",
+		appInstID, appInstMetadata.Type)
+	var ReportInfo = &info.ZInfoMsg{}
+
+	contentType := new(info.ZInfoTypes)
+	*contentType = info.ZInfoTypes_ZiAppInstMetaData
+	ReportInfo.Ztype = *contentType
+	ReportInfo.DevId = *proto.String(zcdevUUID.String())
+	ReportInfo.AtTimeStamp = ptypes.TimestampNow()
+
+	ReportAppInstMetaData := new(info.ZInfoAppInstMetaData)
+
+	ReportAppInstMetaData.Type = info.AppInstMetaDataType(appInstMetadata.Type)
+	ReportAppInstMetaData.Uuid = appInstID
+	ReportAppInstMetaData.Data = appInstMetadata.Data
```

Comment:
We have the size check in zedrouter, correct?
Might make sense to add a comment here that we assume that zedrouter checks that the size does not exceed X to ensure that the info message does not get too large for the controller.

Reply:
Added the check in zedrouter. Added a comment in zedagent about the check.